plugins/modules/network_compliance_workflow_manager.py

```diff
@@ -0,0 +1,1340 @@
+
+            # Update the result with failure status and log the error message
+            self.update_result('failed', False, msg, 'ERROR')
+            return True
+        return False
+
+    def handle_error(self, task_name, mgmt_ip_instance_id_map, failure_reason=None):
+        """
+        Handle error encountered during task execution.
+        Parameters:
+            - task_name (str): Name of the task being performed.
+            - mgmt_ip_instance_id_map (dict): Mapping of management IP addresses to instance IDs.
+            - failure_reason (str, optional): Reason for the failure, if available.
+        Returns:
+            self (object): An instance of the class used for interacting with Cisco Catalyst Center.
+        """
+
+        # If failure reason is provided, include it in the error message
+        if failure_reason:
+            self.msg = "An error occurred while performing {0} on device(s): {1}. The operation failed due to the following reason: {2}".format(
+                task_name, list(mgmt_ip_instance_id_map.keys()), failure_reason)
+        # If no failure reason is provided, generate a generic error message
+        else:
+            self.msg = "An error occurred while performing {0} on device(s): {1}".format(
+                task_name, list(mgmt_ip_instance_id_map.keys()))
+
+        # Update the result with failure status and log the error message
+        self.update_result('failed', False, self.msg, 'ERROR')
+        return self
+
+    def get_compliance_task_status(self, task_id, mgmt_ip_instance_id_map):
+        """
+        This function retrieves the status of compliance check tasks in Cisco Catalyst Center.
+        Parameters:
+            - task_id: The ID of the compliance check task.
+            - mgmt_ip_instance_id_map: A mapping of management IP addresses to instance IDs.
+        Returns:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+        Description:
+            This function continuously checks the status of a compliance check task until completion.
+            It handles various scenarios such as task completion, task failure, or errors during execution.
+            Upon successful completion, it logs the modified compliance response and updates the result accordingly.
+        """
+
+        task_name = 'Run Compliance Check'
+        start_time = time.time()
+
+        while True:
+            response = self.get_task_status(task_id, task_name)
+
+            # Check if response returned
+            if not response:
+                self.msg = 'Error retrieving Task status for the Task {0} with Task Id: {1}'.format(task_name, task_id)
+                self.update_result('failed', False, self.msg, 'ERROR')
+                break
+
+            # Check if the elapsed time exceeds the timeout
+            if self.exit_while_loop(start_time, task_id, task_name, response):
+                break
```

Comment:
Can we print a log message before break?

Reply:
Logging is already happening in the exit_while_loop()

Reply:
I call this method self.update_result('failed', False, msg, 'ERROR') in exit_while_loop.
Which does both logging and updating the result.

def update_result(self, status, changed, msg, log_level, data=None):
    """
    Update the result of the operation with the provided status, message, and log level.
    Parameters:
        - status (str): The status of the operation ('success' or 'failed').
        - changed (bool): Indicates whether the operation caused changes.
        - msg (str): The message describing the result of the operation.
        - log_level (str): The log level at which the message should be logged ('INFO', 'ERROR', 'CRITICAL', etc.).
        - data (dict, optional): Additional data related to the operation result.
    Returns:
        self (object): An instance of the class.
    Note:
        - If the status is 'failed', the 'failed' key in the result dictionary will be set to True.
        - If data is provided, it will be included in the result dictionary.
    """

    # Update the result attributes with the provided values
    self.status = status
    self.result['status'] = status
    self.result['msg'] = msg
    self.result['changed'] = changed

    # Log the message at the specified log level
    self.log(msg, log_level)

    # If the status is 'failed', set the 'failed' key to True
    if status == 'failed':
        self.result['failed'] = True

    # If additional data is provided, include it in the result dictionary
    if data:
        self.result['data'] = data

    return self

Reply:
Cool. Thank you!